From 709b47c3f7d4e41e89ee58a85cd106456f779445 Mon Sep 17 00:00:00 2001
From: "Steven Rostedt (Red Hat)" <rostedt@goodmis.org>
Date: Thu, 17 Sep 2015 15:39:02 -0400
Subject: [PATCH 004/134] event-parse: Fix second increment

Chaos Chen stated that he had a bug in his code; when the usecs overrun, and
1000000 needs to be subtracted from the usecs and added to the seconds, the
seconds need to be incremented not decremented.

Signed-off-by: Steven Rostedt <rostedt@goodmis.org>
---
 event-parse.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/event-parse.c b/event-parse.c
index 9b44125eca46..0a30253e87c8 100644
--- a/event-parse.c
+++ b/event-parse.c
@@ -5244,7 +5244,7 @@ void pevent_print_event(struct pevent *pevent, struct trace_seq *s,
 			/* To avoid usecs larger than 1 sec */
 			if (usecs >= USECS_PER_SEC) {
 				usecs -= USECS_PER_SEC;
-				secs--;
+				secs++;
 			}
 			p = 6;
 		}
-- 
2.5.5

